src/legacy/ui/public/agg_types/agg_param.d.ts

```diff
@@ -20,8 +20,10 @@
 import { AggConfig } from '../vis';
 
 interface AggParam {
+  editorComponent: React.FunctionComponent;
```

Comment:
`React.ComponentType<AggParamEditorProps<unknown>>`

`React.ComponentType` allows for `React.FunctionalComponent` and `React.ComponentClass`, so you can use class components and functional components, which is a bit more flexible (also in most places you use them you really don't care which one it was :D) Also we know which property type we require here (except the `T` prameter so leave that at `unknown`), so we should use it, i.e. `AggParamEditorProps`.